[REQUEST] Remove deprecated PyTorch code

**Is your feature request related to a problem? Please describe.**
DeepSpeed uses code that's deprecated. This causes problems for libraries built on deepspeed when testing.

**Describe the solution you'd like**
```
/usr/local/lib/python3.10/dist-packages/deepspeed/comm/torch.py:171: in get_global_rank
    return torch.distributed.distributed_c10d._get_global_rank(group, group_rank)
```
should instead use `get_global_rank` as recommended by the warning
```
E       UserWarning: torch.distributed.distributed_c10d._get_global_rank is deprecated please use torch.distributed.distributed_c10d.get_global_rank instead
```

--- a/deepspeed/comm/__init__.py
+++ b/deepspeed/comm/__init__.py
@@ -19,7 +19,7 @@ if older_torch():
         return group.WORLD
 
     def get_global_rank(group, group_rank):
-        is hasattr(torch.distributed.distributed_c10d, "get_global_rank"):
+        if hasattr(torch.distributed.distributed_c10d, "get_global_rank"):
             from torch.distributed.distributed_c10d import get_global_rank as _get_global_rank
         else:
             from torch.distributed.distributed_c10d import _get_global_rank

--- a/deepspeed/comm/__init__.py
+++ b/deepspeed/comm/__init__.py
@@ -19,7 +19,10 @@ if older_torch():
         return group.WORLD
 
     def get_global_rank(group, group_rank):
-        from torch.distributed.distributed_c10d import _get_global_rank
+        is hasattr(torch.distributed.distributed_c10d, "get_global_rank"):
+            from torch.distributed.distributed_c10d import get_global_rank as _get_global_rank
+        else:
+            from torch.distributed.distributed_c10d import _get_global_rank
         return _get_global_rank(group, group_rank)
 
     def allgather_fn(output_tensor, input_tensor, group=None, async_op=False):

--- a/deepspeed/comm/torch.py
+++ b/deepspeed/comm/torch.py
@@ -168,7 +168,11 @@ class TorchBackend(Backend):
         return torch.distributed.new_group(ranks)
 
     def get_global_rank(self, group, group_rank):
-        return torch.distributed.distributed_c10d._get_global_rank(group, group_rank)
+        is hasattr(torch.distributed.distributed_c10d, "get_global_rank"):
+            from torch.distributed.distributed_c10d import get_global_rank as _get_global_rank
+        else:
+            from torch.distributed.distributed_c10d import _get_global_rank
+        return _get_global_rank(group, group_rank)
 
     def get_world_group(self):
         return torch.distributed.group.WORLD

--- a/deepspeed/comm/torch.py
+++ b/deepspeed/comm/torch.py
@@ -168,7 +168,7 @@ class TorchBackend(Backend):
         return torch.distributed.new_group(ranks)
 
     def get_global_rank(self, group, group_rank):
-        is hasattr(torch.distributed.distributed_c10d, "get_global_rank"):
+        if hasattr(torch.distributed.distributed_c10d, "get_global_rank"):
             from torch.distributed.distributed_c10d import get_global_rank as _get_global_rank
         else:
             from torch.distributed.distributed_c10d import _get_global_rank

